extension-iac/src/iac_android.cpp

```diff
@@ -284,59 +198,36 @@ dmExtension::Result InitializeIAC(dmExtension::Params* params)
 
 dmExtension::Result FinalizeIAC(dmExtension::Params* params)
 {
-    if (params->m_L == g_IAC.m_Listener.m_L && g_IAC.m_Listener.m_Callback != LUA_NOREF) {
-        dmScript::Unref(g_IAC.m_Listener.m_L, LUA_REGISTRYINDEX, g_IAC.m_Listener.m_Callback);
-        dmScript::Unref(g_IAC.m_Listener.m_L, LUA_REGISTRYINDEX, g_IAC.m_Listener.m_Self);
-        g_IAC.m_Listener.m_L = 0;
-        g_IAC.m_Listener.m_Callback = LUA_NOREF;
-        g_IAC.m_Listener.m_Self = LUA_NOREF;
+    if (params->m_L == dmScript::GetCallbackLuaContext(g_IAC.m_Listener)) {
```

Comment:
Fyi, this is one of the symptoms we see of the "shared_state == 0".

Reply:
👍